src/main/java/domain/board/BoardDao.java

```diff
@@ -0,0 +1,145 @@
+			System.err.println("연결 오류:" + e.getMessage());
+			e.printStackTrace();
+		}
+
+		return con;
+	}
+
+	// 드라이버 연결해제
+	public void closeConnection(Connection con) {
+		try {
+			if (con != null)
+				con.close();
+		} catch (SQLException e) {
+			System.err.println("con 오류:" + e.getMessage());
+		}
+	}
+
+	public void clearBoardDb() throws SQLException {
+		String query = "DELETE FROM board";
+		PreparedStatement pstmt = getConnection().prepareStatement(query);
+		pstmt.executeUpdate();
+	}
+
+	public void saveBoard(Board board) throws SQLException {
+		Connection connection = getConnection();
+		for (Rank rank : board.getRanks()) {
+			savePiece(rank, connection);
+		}
+	}
+
+	private void savePiece(Rank rank, Connection connection) throws SQLException {
+		for (Piece piece : rank.getPieces()) {
+			String query = "INSERT INTO board VALUES (?, ?)";
+			PreparedStatement pstmt = connection.prepareStatement(query);
+			String position =
+				piece.getPosition().getColumn().getColumnName() + String.valueOf(piece.getPosition().getRow());
+
+			pstmt.setString(1, position);
+			pstmt.setString(2, piece.showSymbol());
+			pstmt.executeUpdate();
+		}
+	}
+
+	public void clearTurn() throws SQLException {
+		String query = "DELETE FROM TURN";
+		PreparedStatement pstmt = getConnection().prepareStatement(query);
+		pstmt.executeUpdate();
+	}
+
+	public void saveTurn(Team turn) throws SQLException {
+		String query = "INSERT INTO turn VALUES (?)";
+		PreparedStatement pstmt = getConnection().prepareStatement(query);
+		pstmt.setString(1, turn.getName());
+		pstmt.executeUpdate();
+	}
+
+	public Board loadBoard() throws SQLException {
+		List<Rank> ranks = new ArrayList<>();
+		Connection connection = getConnection();
+
+		for (int i = FIRST_RANK_INDEX; i <= LAST_RANK_INDEX; i++) {
+			ranks.add(loadRank(i, connection));
+		}
+		return new Board(ranks);
+	}
+
+	private Rank loadRank(int rankIndex, Connection connection) throws SQLException {
+		List<Piece> pieces = new ArrayList<>();
+		String query = "SELECT * FROM board";
+		PreparedStatement pstmt = connection.prepareStatement(query);
+		ResultSet rs = pstmt.executeQuery();
+		char rankOfPiece;
+		int pieceRank;
+
+		while (rs.next()) {
+			rankOfPiece = rs.getString("piece_position").charAt(RANK_INDEX);
+			pieceRank = Integer.parseInt(String.valueOf(rankOfPiece));
+			if (pieceRank == rankIndex) {
+				pieces.add(createPiece(rs));
+			}
+		}
+		return new Rank(pieces);
+	}
+
+	private Piece createPiece(ResultSet rs) throws SQLException {
+		Position position = Position.of(rs.getString("piece_position"));
+		return PieceCreator.of(rs.getString("symbol")).getPieceCreator().apply(position);
+	}
+
+	public Team loadTurn() throws SQLException {
+		String query = "SELECT * FROM turn";
+		PreparedStatement pstmt = getConnection().prepareStatement(query);
+		ResultSet rs = pstmt.executeQuery();
+		if (rs.next()) {
+			return Team.of(rs.getString("game_turn"));
+		}
+		return null;
```

Comment:
Optional을 반환하는 형태가 좋아보입니다.

Reply:
```suggestion
		throw new IllegalArgumentException("데이터 베이스에 저장된 Turn이 없습니다.");
```
리뷰어님께서 null로 인해 발생하는 문제때문에 Optional의 사용을 추천해주셨다고 생각했습니다. 그래서 DB가 비었을 때 새로운 exception이 발생하도록 수정해보았습니다. 
혹시 리뷰어님이 의도하신 피드백이 아니라면 다시 피드백 주시면 다시 생각해보고 적용해보겠습니다. :)

Reply:
리뷰어님이 위에서 제안해주신 `인터페이스로 분리하여 중복되는 코드를 재활용`하면서 DB에 저장된 Turn을 반환받는 부분의 `return null;`이 자연스럽게 사라졌습니다. :)

추가로 만약 DB에 저장된 정보가 없으면 exception으로 인해 프로그램이 종료되는 것이 아닌, ErrorMessage를 출력하고 게임이 강제로 종료되지 않도록 구현해봤습니다. :)